### PR TITLE
Move providers to pkg/provider and refactor interface

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/thegreenwebfoundation/grid-intensity-go/issues/44

This moves the Carbon Intensity UK and ElectricityMap providers to `pkg/providers`. The next PR will migrate the other 2 providers.

It also changes the interface from

```
GetCarbonIntensity(ctx context.Context, region string) (float64, error)
```

to 

```
GetCarbonIntensity(ctx context.Context, region string) ([]CarbonIntensity, error)
```

Returning a struct means we can provide more context about the emissions data. Like whether it is marginal or average and units being used.

The changes are quite big but the file structure changes and `pkg/provider/provider.go` are probably the most interesting parts.

### CODE REVIEW

File: pkg/provider/provider.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	// Supported metric types.
+	AbsoluteMetricType = "absolute"
+	RelativeMetricType = "relative"
+
+	// Supported units.
+	GramsCO2EPerkWh = "gCO2e per kWh"
+
+	// Supported providers
+	CarbonIntensityOrgUK = "carbonintensity.org.uk"
+	ElectricityMap       = "electricitymap.org"
```
Comment (rossf7) [inline pkg/provider/provider.go:22]: These are the providers supported so far with the new structure.

File: pkg/provider/provider.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+type CarbonIntensity struct {
```
Comment (rossf7) [inline pkg/provider/provider.go:25]: This is the new data structure for returning the data.

File: cmd/exporter.go [hunk 3]
```diff
@@ -86,41 +85,45 @@ the grid is greener or at locations where carbon intensity is lower.
...
 		if err != nil {
 			return nil, err
 		}
-		units = "gCO2 per kWh"
-	case electricitymap.ProviderName:
+	case provider.ElectricityMap:
 		token := os.Getenv(electricityMapAPITokenEnvVar)
 		if token == "" {
-			return nil, fmt.Errorf("please set the env variable %q", electricityMapAPITokenEnvVar)
+			return nil, fmt.Errorf("%q env var must be set", electricityMapAPITokenEnvVar)
```
Comment (mrchrisadams) [inline cmd/exporter.go:117]: @rossf7  - can we please make an issue to revisit all the error and command line copy?

I feel I can share my reckons on each PR, but there must be some good guidelines online somewhere we can refer to in future when reviewing PRs, so there's something that isn't as subjective as me giving rando comments  all the time.

**Comment by mrchrisadams** [comment]
I've looked over this, and I'm happy with it @rossf7  - feel free to merge away

File: cmd/exporter.go (hunk 3 above)
Comment (rossf7) [reply cmd/exporter.go:117]: @mrchrisadams Yes, I changed this for consistency with other providers but my bad it makes the PR harder to review.

I've created https://github.com/thegreenwebfoundation/grid-intensity-go/issues/51 to revisit this. 

It also might make sense to use an external library like `juju/errors`
https://github.com/juju/errors

**Pull request merged**